src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -281,6 +282,140 @@ public int Y2
             Assert.Equal(expectedText, VisualStudio.Editor.GetText());
         }
 
+        // DO NOT SKIP THIS TEST. IT MUST PASS.
+        [WpfTheory]
+        [InlineData(FixAllScope.Project)]
+        [InlineData(FixAllScope.Solution)]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
+        [WorkItem(33507, "https://github.com/dotnet/roslyn/issues/33507")]
+        public void FixAllOccurrencesIgnoresGeneratedCode(FixAllScope scope)
+        {
+            var markup = @"
+using System;
+using $$System.Threading;
+
+class C
+{
+    public IntPtr X1 { get; set; }
+}";
+            var expectedText = @"
+using System;
+
+class C
+{
+    public IntPtr X1 { get; set; }
+}";
+            var generatedSourceMarkup = @"// <auto-generated/>
+using System;
+using $$System.Threading;
+
+class D
+{
+    public IntPtr X1 { get; set; }
+}";
+            var expectedGeneratedSource = @"// <auto-generated/>
+using System;
+
+class D
+{
+    public IntPtr X1 { get; set; }
+}";
+
+            MarkupTestFile.GetPosition(generatedSourceMarkup, out var generatedSource, out int generatedSourcePosition);
+
+            VisualStudio.SolutionExplorer.AddFile(new ProjectUtils.Project(ProjectName), "D.cs", generatedSource, open: false);
+
+            // Switch to the main document we'll be editing
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), "Class1.cs");
+
+            // Verify that applying a Fix All operation does not change generated files
+            SetUpEditor(markup);
+            VisualStudio.WaitForApplicationIdle(CancellationToken.None);
+            VisualStudio.Editor.InvokeCodeActionList();
+            VisualStudio.Editor.Verify.CodeAction(
+                "Remove Unnecessary Usings",
+                applyFix: true,
+                fixAllScope: scope);
+
+            Assert.Equal(expectedText, VisualStudio.Editor.GetText());
+
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), "D.cs");
+            Assert.Equal(generatedSource, VisualStudio.Editor.GetText());
+
+            // Verify that a Fix All in Document in the generated file still does nothing
```

Comment:
This seems wrong. We should either not be offering a FixAll in document in generated files OR if we do offer it then it must apply the Fix all instances in the document.

Reply:
This is a regression test showing preservation of 15.0 behavior in 16.0.

Reply:
Well, the behavior itself seems broken - offering users a fix that does nothing, and is by design, is neither intuitive nor a good user experience. I am fine with your change as that just reverts to 15.0 behavior, but we may want to discuss just hiding the FixAll in document option for code fixes offered in generated files in 16.1

Reply:
Can we add a comment that we believe this behavior is wrong? I strongly detest tests that assert known bad behavior without clear comments tothat effect, because it's confusing as hell when you're trying to fix something, "break" a test because you did the right thing, and then ultimately figure it out that way.

Reply:
What would it take to just fix the behavior instead?

Reply:
> What would it take to just fix the behavior instead?

That would be for 16.1 at the earliest

Reply:
> Can we add a comment that we believe this behavior is wrong? 

Comments have been added for clarity.